arctic/store/_ndarray_store.py

```diff
@@ -163,6 +160,41 @@ def get_info(self, version):
         ret['rows'] = int(version['up_to'])
         return ret
 
+    def chunked_read(self, arctic_lib, version, symbol, date_range, read_preference=None):
+        collection = arctic_lib.get_top_level_collection()
+        if read_preference:
+            collection = collection.with_options(read_preference=read_preference)
+        return self._do_chunked_read(collection, version, symbol, date_range)
+
+    def _do_chunked_read(self, collection, version, symbol, date_range):
+        segment_count = version.get('segment_count', None)
+
+        spec = {'symbol': symbol,
+                'parent': version.get('base_version_id', version['_id']),
+                }
+
+        if date_range is not None:
+            spec['segment'] = date_range.mongo_query()
```

Comment:
Would something like:

- Chunked by year
- TS containing: 2015/01/01  ;  2015/06/01 ; 2016/01/01
- Query 2015/05/30 -> 2016/01/02 should return two rows 

work?